sematic/abstract_future.py

```diff
@@ -102,25 +102,44 @@ class AbstractFuture(abc.ABC):
     Parameters
     ----------
     calculator: AbstractCalculator
-        The calculator this is a future of
-    kwargs: typing.Dict[str, typing.Any]
+        The calculator this is a future of.
+
+    kwargs: Dict[str, Any]
         The input arguments to the calculator. Can be concrete values or other futures.
+
+    inline: bool
+        When using the `CloudResolver`, whether the instrumented function should be
+        executed inside the same process and worker that is executing the `Resolver`
+        itself.
+
+    original_future_id: Optional[str]
+        The id of the original future this future was cloned from, if any.
+
+    resource_requirements: Optional[ResourceRequirements]
+        When using the `CloudResolver`, specifies what special execution resources the
+        function requires. Defaults to `None`.
+
+    retry_settings: Optional[RetrySettings]
+        Specifies in case of which Exceptions the function's execution should be retried,
+        and how many times. Defaults to `None`.
```

Comment:
We follow the Numpy docstring conventions which do not have blank lines between parameters. See https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html